pyproject.toml

```diff
@@ -24,7 +24,7 @@ include = [
 [tool.poetry.dependencies]
 python = ">=3.8,<3.12"
 great-expectations = "^0.18.17"
-pydantic = "<3"
+pydantic = ">=2.8.1,<3"
```

Comment:
This now requires pydantic v2.
This soft requirement before is what caused us to use the pydantic compatibility layer which is what led to the recent incident.